frontend/src/reducers/productsReducer.js

```diff
@@ -0,0 +1,29 @@
+//import ACTION
+import { FETCH_PRODUCTS } from "../actions/productsAction";
+import { SET_QUERY } from "../actions/queryAction";
+
+const initialState = {
+  query: [],
```

Comment:
I like that you put the `query` inside the reducer, but (unless I'm reading this wrong) in this case it looks like it's unnecessary because you're never using it, though you're updating.